### PR TITLE
add golangci-lint action

### PR DESCRIPTION
**Pull request by acgreek**

### CODE REVIEW

**Comment by lesismal** [comment]
Already pushed to master and `httpcli` will be dropped. I didn't merge `httpcli` to `master` but copy and cover master and push it due to many temporary changes and no specific logs.

Please push this directly to master.

**Comment by acgreek** [comment]
@googlebot I consent

File: nbhttp/websocket/conn.go [hunk 1]
```diff
@@ -139,10 +139,11 @@ func (c *Conn) WriteMessage(messageType MessageType, data []byte) error {
 		cw := compressWriter(w, c.compressionLevel)
 		_, err := cw.Write(data)
 		if err != nil {
-			return err
+			compress = false
```
Comment (lesismal) [inline nbhttp/websocket/conn.go:142]: Do not return err, but continue to send data without compression, then we do not need to judge the return value of WriteMessage, else we don't know whether the connection has been closed by the framework.

Comment (acgreek) [reply nbhttp/websocket/conn.go:142]: this is merged in from later commits

**Comment by lesismal** [comment]
Thank you for so much work!

**Pull request merged**